art/estimators/certification/derandomized_smoothing/derandomized_smoothing_tensorflow.py

```diff
@@ -230,25 +125,44 @@ def __call__(
         """
         return self.forward(x=x, column_pos=column_pos)
 
-    def certify(self, preds: np.ndarray, size_to_certify: int) -> np.ndarray:
+    def certify(
+        self, preds: tf.Tensor, size_to_certify: int, label: Union[np.ndarray, tf.Tensor]
+    ) -> Tuple[tf.Tensor, tf.Tensor, tf.Tensor]:
         """
         Checks if based on the predictions supplied the classifications over the ablated datapoints result in a
         certified prediction against a patch attack of size size_to_certify.
 
         :param preds: The cumulative predictions of the classifier over the ablation locations.
         :param size_to_certify: The size of the patch to check against.
+        :param label: Ground truth labels
         :return: Array of bools indicating if a point is certified against the given patch dimensions.
         """
-        indices = np.argsort(-preds, axis=1, kind="stable")
-        values = np.take_along_axis(np.copy(preds), indices, axis=1)
+        result = tf.math.top_k(preds, k=2)
+
+        top_predicted_class, second_predicted_class = result.indices[:, 0], result.indices[:, 1]
+        top_class_counts, second_class_counts = result.values[:, 0], result.values[:, 1]
 
-        num_affected_classifications = size_to_certify + self.ablation_size - 1
+        certs = (top_class_counts - second_class_counts) > 2 * (size_to_certify + self.ablation_size - 1)
 
-        margin = values[:, 0] - values[:, 1]
+        tie_break_certs = (
+            (top_class_counts - second_class_counts) == 2 * (size_to_certify + self.ablation_size - 1)
+        ) & (top_predicted_class < second_predicted_class)
+        cert = tf.math.logical_or(certs, tie_break_certs)
 
-        certs = margin > 2 * num_affected_classifications
-        tie_break_certs = (margin == 2 * num_affected_classifications) & (indices[:, 0] < indices[:, 1])
-        return np.logical_or(certs, tie_break_certs)
+        # NB, newer versions of pylint do not require the disable.
+        if label.ndim > 1:
+            cert_and_correct = cert & (
+                tf.math.argmax(label, axis=1)
+                == tf.cast(  # pylint: disable=E1120, E1123
```

Comment:
Why are these warnings disabled?
```suggestion
                == tf.cast(
```